src/main/java/io/github/nucleuspowered/nucleus/modules/blacklist/listeners/BlacklistListener.java

```diff
@@ -0,0 +1,97 @@
+import org.spongepowered.api.item.ItemType;
+import org.spongepowered.api.item.ItemTypes;
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.transaction.SlotTransaction;
+
+import java.util.Optional;
+
+public class BlacklistListener extends ListenerBase {
+
+    @Listener
+    public void onPlayerChangeHeldItem(ChangeInventoryEvent.Held event, @First Player player) {
+        for (SlotTransaction transaction : event.getTransactions()) {
+            if (checkForBlacklist(player, transaction.getFinal().createStack().getItem().getId())) {
+                transaction.setCustom(ItemStack.builder().itemType(ItemTypes.DIRT).quantity(1).build());
+            }
+        }
+    }
+
+    @Listener
+    public void onPlayerChangeEquipment(ChangeInventoryEvent.Equipment event, @First Player player) {
+        for (SlotTransaction transaction : event.getTransactions()) {
+            if (checkForBlacklist(player, transaction.getFinal().createStack().getItem().getId())) {
+                transaction.setCustom(ItemStack.builder().itemType(ItemTypes.DIRT).quantity(1).build());
+            }
+        }
+    }
+
+    @Listener
+    public void onPlayerDropItem(DropItemEvent.Dispense event, @First Player player) {
+        event.filterEntities(e -> {
+            if (e.get(RepresentedItemData.class).isPresent()) {
+                RepresentedItemData itemData = e.get(RepresentedItemData.class).get();
+                return !checkForBlacklist(player, itemData.item().get().getType().getId());
+            }
+            return false;
+        });
+    }
+
+    @Listener
+    public void onPlayerPickupItem(ChangeInventoryEvent.Pickup event, @First Player player) {
+        for (SlotTransaction transaction : event.getTransactions()) {
+            if (checkForBlacklist(player, transaction.getFinal().createStack().getItem().getId())) {
+                transaction.setCustom(ItemStack.builder().itemType(ItemTypes.DIRT).quantity(1).build());
+            }
+        }
+    }
+
+    @Listener
+    public void onPlayerInteractBlock(InteractBlockEvent event, @First Player player) {
+        event.setCancelled(checkForBlacklist(player, event.getTargetBlock().getState().getType().getId()));
+    }
+
+    @Listener
+    public void onPlayerPlaceBlock(ChangeBlockEvent.Place event, @First Player player) {
+        for (Transaction<BlockSnapshot> transaction : event.getTransactions()) {
+            event.setCancelled(checkForBlacklist(player, transaction.getFinal().getState().getType().getId()));
+        }
+    }
+
+    @Listener
+    public void onPlayerBreakBlock(ChangeBlockEvent.Break event, @First Player player) {
+        for (Transaction<BlockSnapshot> transaction : event.getTransactions()) {
+            event.setCancelled(checkForBlacklist(player, transaction.getOriginal().getState().getType().getId()));
+        }
+    }
+
+    private boolean checkForBlacklist(Player player, String id) {
```

Comment:
Bit concerned that if multiple blocks are to be confiscated, this will get spammy.